src/status_im/ui/screens/chat/message/message.cljs

```diff
@@ -30,11 +31,14 @@
 (defview message-content-command
   [command-message]
   (letsubs [id->command [:chats/id->command]]
-    (if-let [command (commands-receiving/lookup-command-by-ref command-message id->command)]
-      (commands/generate-preview command command-message)
-      (if-let [extension-id (get-in command-message [:content :params :extension-id])]
+    (let [{:keys [type] :as command} (commands-receiving/lookup-command-by-ref command-message id->command)
+          extension-id (get-in command-message [:content :params :extension-id])]
+      (if (and extension-id (or (not type) (and type (satisfies? protocol/Extension type)
+                                                (not= extension-id (protocol/extension-id type)))))
         [install-extension-message extension-id (:outgoing command-message)]
-        [react/text (str "Unhandled command: " (-> command-message :content :command-path first))]))))
+        (if command
+          (commands/generate-preview command command-message)
+          [react/text (str "Unhandled command: " (-> command-message :content :command-path first))])))))
```

Comment:
Is this for backward compatibility? Older messages w/o extension id embedded? If so do we get the nice upgrade message if the id is provided but not the right one?

Reply:
no, this is for all nonsupported commands doesn't matter if it an extension or not.  We don't have a nice upgrade message, because we don't know if it's an upgrade or different extension, there will be standard install extension message

Reply:
done